app.rb

```diff
@@ -0,0 +1,118 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    return puts 'No books found' if @books.empty?
+
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    return puts 'No people found' if @people.empty?
+
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    if person_type == '1'
+      create_student(age, name)
+    else
+      create_teacher(age, name)
+    end
+
+    puts 'Person created successfully'
+  end
+
+  def create_student(age, name)
```

Comment:
- [ ] When I tried to create a student I had an error because of the number of arguments, please correct the number of arguments to avoid issues 👇🏽 

![image](https://github.com/hasnatali1947/-OOP_school_library/assets/58642949/6ad10204-c63d-48e4-bb30-c99b828ca6d4)